plugins/database/cassandra/connection_producer.go

```diff
@@ -269,6 +230,50 @@ func (c *cassandraConnectionProducer) createSession(ctx context.Context) (*gocql
 	return session, nil
 }
 
+func getSslOpts(certBundle *certutil.CertBundle, minTLSVersion string, insecureSkipVerify bool) (*gocql.SslOptions, error) {
+	var tlsConfig *tls.Config
+	if certBundle != nil {
+		if certBundle.Certificate == "" && certBundle.PrivateKey != "" {
+			return nil, fmt.Errorf("found private key for TLS authentication but no certificate")
+		}
+		if certBundle.Certificate != "" && certBundle.PrivateKey == "" {
+			return nil, fmt.Errorf("found certificate for TLS authentication but no private key")
+		}
+
+		parsedCertBundle, err := certBundle.ToParsedCertBundle()
+		if err != nil {
+			return nil, fmt.Errorf("failed to parse certificate bundle: %w", err)
+		}
+
+		tlsConfig, err := parsedCertBundle.GetTLSConfig(certutil.TLSClient)
```

Comment:
Should this be using `=` instead of `:=` when assigning to `tlsConfig`? I think using `:=` here would result in a nil pointer dereference down on L256.

Reply:
Nice catch!

Reply:
Ya, was thinking it'd be nil pointer dereference if we don't enter that else branch (since certBundle != nil). In any case, I agree that the correct `tlsConfig` won't be set due to the scoping.

Reply:
I think you're right. Since both variables exist before this line they are shadowed by this call when using `:=`. Or at least that's what my IDE is telling me.